x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java

```diff
@@ -156,13 +158,30 @@ protected void doExecute(Task submitTask, GetStackTracesRequest request, ActionL
         responseBuilder.setCustomCostPerCoreHour(request.getCustomCostPerCoreHour());
         Client client = new ParentTaskAssigningClient(this.nodeClient, transportService.getLocalNode(), submitTask);
         if (request.getIndices() == null) {
-            searchProfilingEvents(client, request, submitListener, responseBuilder);
+            searchProfilingEvents(submitTask, client, request, submitListener, responseBuilder);
         } else {
-            searchGenericEvents(client, request, submitListener, responseBuilder);
+            searchGenericEvents(submitTask, client, request, submitListener, responseBuilder);
+        }
+    }
+
+    /**
+     * Checks whether a task has been cancelled and notifies the provided listener if required.
+     * @param task The task to check. May be a cancelable task.
+     * @param listener Listener to notify.
+     * @return <code>true</code> iff the task has been cancelled. Callers must terminate as early as possible.
+     */
+    private boolean mayNotifyOfCancellation(Task task, ActionListener<GetStackTracesResponse> listener) {
```

Comment:
nit: Maybe move this function closer to where it is actually used.

Reply:
Good point. I think I'll keep this for now as we should restructure that action anyway soon and break it into smaller pieces.